codespell_lib/_codespell.py

```diff
@@ -228,7 +229,8 @@ def open_with_chardet(self, filename: str) -> Tuple[List[str], str]:
                     break
         self.encdetector.close()
         encoding = self.encdetector.result["encoding"]
-        assert encoding is not None  # noqa: S101
+        if not encoding:
+            encoding = locale.getpreferredencoding(False)
```

Comment:
Thinking about this again (sorry I didn't realize this earlier!) -- I am a bit torn on this. On the one hand this might tend to make things work for people. On the other, this means behavior on one machine will differ from another. That makes bugs and reproducibility harder for people, especially because this will be silent. WDYT about just setting `encoding = 'utf8'` as the fallback instead?

Reply:
Good point regarding the different behavior on different machines, but forcing a default encoding could lead to problems and I think it might make more sense to let Python cope with it. How about:
Instead of setting a encoding in before opening the file (either using a default or `getpreferredencoding`) we keep the old behavior. If we pass `None` as `encoding` to `open` Python will take care of it, just like before.
But instead of returning a possible `None` encoding, we return either `encoding` or `f.encoding`, because the latter will contain the encoding detected by Python after successfully opening the file.
```
>>> f = open("empty_file")
>>> f.encoding
'UTF-8'
>>> f = open("empty_file", encoding="ASCII")
>>> f.encoding
'ASCII'
```
This will have the same behavior like now, but letting Python take care of selecting the default encoding.

Reply:
> it would be really nice to cover this line in a test

Done! Writing problematic bytes into the file isn't really necessary though, because the bug can be triggered by an empty file :)

Reply:
>  If we pass None as encoding to open Python will take care of it, just like before.

Works for me!

Reply:
Done!
I also fixed the formatting error reported by black. There also were some other errors reported by ruff, but in lines I didn't even touch. I don't need to worry about them, do I?

Reply:
If you rebase, ruff errors should disappear.

Reply:
Ah, I already did and now they disappeared. Thanks :)